src/jit/lir.cpp

```diff
@@ -195,7 +195,14 @@ void LIR::Use::ReplaceWith(Compiler* compiler, GenTree* replacement)
     *m_edge = replacement;
     if (!IsDummyUse() && m_user->IsCall())
     {
-        compiler->fgFixupArgTabEntryPtr(m_user, replacedNode, replacement);
+        GenTreeCall* call = m_user->AsCall();
+        const bool isCallArg = (replacement != call->gtControlExpr) && ((call->gtCallType != CT_INDIRECT) ||
```

Comment:
I think this condition is a bit brittle and has the knowledge of the call node state which could change in future.

Instead here is a suggestion:

How about having a new method gtGetArgEntryByNodeIfAny() that will not assert even if it could find an arg table entry.  Existing method gtArgEntryByNode() could be implemented in terms of gtGetArgEntryByNodeIfAny().

Before replacing the node, we will check to see whether we are replacing an arg

```
fgArgTabEntryPtr fp = compiler->gtArgEntryByNodeIfAny(call, replacedNode);
...
// replace the node
..
if (fp != nullptr)
{
    // We need to implement a new overload method fgFixupArgTabEntry(fgArgTabEntryPtr fp, GenTreePtr oldArg, GenTreePtr newArg).
   // Existing method fgFixupArgTabEntryPtr(GenTreePtr parentCall, GenTreePtr oldArg, GenTreePtr newArg) could be implemented in terms of above overload.
    compiler->fgFixupArgTabEntryPtr(fp, replacedNode, replacement);
}
```